### ISSUE TITLE
Cannot reproduce the results in Table 11 on C-Town dataset

### ISSUE THREAD

**Issue by zhiyuan25** [DiTEC-project/gnn-pressure-estimation#5]
Hello author, I am currently conducting further research based on your paper "Graph Neural Networks for Pressure Estimation in Water Distribution Systems". This is a very great job. Thank you for your contribution in this field and for making the code public.

I am attempting to conduct further research based on your study. I have decided to start with the small public dataset c-town and attempt to achieve the same results as those shown in Table 11. According to the description in Section 6.4 "Baseline Comparison on Benchmark WDNs", I conducted an experiment, but the results obtained were not consistent with the data in the original text. I would like to ask you if I made any mistakes in any of my steps. My steps are as follows:

1. Modify the "num_scenarios = 100" in the "configurations/v7.1/ctown_7v1__EPYNET_config.ini" file under code/test/gnn-pressure-estimation to 1000 to ensure consistency with the number of scenarios in the paper c_town.
2. Generate the dataset using the following command and created c_town.zip:
`python gnn_pressure_estimation/scenegenv7.py --executors 2 --batch_size 500 --gen_demand --gen_res_total_head --config configs/v7.1/ctown_7v1__EPYNET_config.ini`
3. Perform training using the following command:
`python gnn_pressure_estimation/train.py  --dataset_paths datasets/ctown.zip --input_paths inputs/ctown.inp --test_data_path datasets/ctown.zip --test_input_path inputs/ctown.inp --model gatres_small --lr 0.0005 --weight_decay 0.000001 --batch_size 32 --epochs 2000 --patience 250 --feature pressure --mask_rate 0.95 --num_trains 1000 --do_test True`

My result:
**MAE: 6.4406 +/- 0.0556**
```
GATResMeanConv(
  (lin0): Linear(1, 32, bias=True)
  (blocks): ModuleList(
    (0-14): 15 x GResBlockMeanConv(
      (conv1): GATConv(32, 32, heads=2)
      (conv2): GATConv(64, 32, heads=1)
      (mean_conv): SimpleConv()
    )
  )
  (lin1): Linear(32, 1, bias=True)
)
Model parameters:  65857
################################################################################
args list:
model = gatres_small
lr = 0.0005
weight_decay = 1e-06
epochs = 2000
mask_rate = 0.95
dataset_paths = ['.../gnn-pressure-estimation/datasets/ctown.zip']
input_paths = ['.../gnn-pressure-estimation/inputs/ctown.inp']
do_test = True
test_data_path = /.../gnn-pressure-estimation/datasets/ctown.zip
test_input_path = /.../gnn-pressure-estimation/inputs/ctown.inp
test_removal = keep_junction
feature = pressure
variant = 20251027_2108
model_name = GATResMeanConv_small_znorm_15b_32c
criterion = mse
num_trains = 1000
batch_size = 32
use_data_batch = False
use_data_edge_attrs = None
patience = 250
min_delta = 0.0001
train_val_removal = keep_junction
device = cuda
use_gradient_clipping = False
percentile = 10.0
scheduler = None
scheduler_patience = 2
norm_type = znorm
log_method = None
log_gradient = True
project_name = test_project
save_path = experiments_logs/test_args/fun_test
num_test_trials = 10
model_path = experiments_logs\simple_test\gatres_znorm\best_GATResMeanConv_znorm_20235922.pth
......
Epoch: 001, train loss: 1.0169, val_loss: 0.8934, val_error: 0.4480, val_0.1: 0.2232, val_corr: 0.4253, val_r2: 0.1811, val_mae: 21.0260, val_rmse: 26.0530, val_mynse: 0.1096, 
Epoch: 005, train loss: 0.7455, val_loss: 0.7348, val_error: 0.3782, val_0.1: 0.3082, val_corr: 0.5178, val_r2: 0.2682, val_mae: 17.9986, val_rmse: 23.6269, val_mynse: 0.2660, 
Epoch: 010, train loss: 0.7204, val_loss: 0.7037, val_error: 0.3587, val_0.1: 0.3115, val_corr: 0.5455, val_r2: 0.2978, val_mae: 17.6207, val_rmse: 23.1210, val_mynse: 0.2970, 
Epoch: 015, train loss: 0.6922, val_loss: 0.6869, val_error: 0.3573, val_0.1: 0.3171, val_corr: 0.5614, val_r2: 0.3154, val_mae: 17.3324, val_rmse: 22.8430, val_mynse: 0.3142, 
Epoch: 020, train loss: 0.7002, val_loss: 0.6914, val_error: 0.3619, val_0.1: 0.3232, val_corr: 0.5613, val_r2: 0.3151, val_mae: 17.3035, val_rmse: 22.9192, val_mynse: 0.3096, 
......
Epoch: 1980, train loss: 0.1241, val_loss: 0.1255, val_error: 0.1165, val_0.1: 0.6728, val_corr: 0.9366, val_r2: 0.8773, val_mae: 6.6011, val_rmse: 9.7552, val_mynse: 0.8749, 
Epoch: 1985, train loss: 0.1244, val_loss: 0.1121, val_error: 0.1160, val_0.1: 0.6843, val_corr: 0.9424, val_r2: 0.8880, val_mae: 6.3841, val_rmse: 9.2254, val_mynse: 0.8879, 
Epoch: 1990, train loss: 0.1167, val_loss: 0.1152, val_error: 0.1165, val_0.1: 0.6898, val_corr: 0.9407, val_r2: 0.8850, val_mae: 6.3323, val_rmse: 9.3453, val_mynse: 0.8849, 
Epoch: 1995, train loss: 0.1390, val_loss: 0.1241, val_error: 0.1195, val_0.1: 0.6716, val_corr: 0.9367, val_r2: 0.8774, val_mae: 6.6558, val_rmse: 9.6926, val_mynse: 0.8762, 
Epoch: 2000, train loss: 0.1260, val_loss: 0.1196, val_error: 0.1246, val_0.1: 0.6745, val_corr: 0.9399, val_r2: 0.8834, val_mae: 6.5781, val_rmse: 9.5295, val_mynse: 0.8807, 
......
test_loss: 0.1206 +/- 0.0023, test_loss_sensor: 0.1194 +/- 0.0016, test_error: 0.1191 +/- 0.0013, test_0.1: 0.6898 +/- 0.0027, test_corr: 0.9381 +/- 0.0013, test_r2: 0.8801 +/- 0.0024, test_mae: 6.4406 +/- 0.0556, test_rmse: 9.5586 +/- 0.0974, test_mynse: 0.8795 +/- 0.0024, test_time: 23.2082 +/- 0.1088, test_throughput: 44123.2756 +/- 206.9870, test_error_sensor: 0.1183 +/- 0.0010, test_0.1_sensor: 0.6902 +/- 0.0011, test_corr_sensor: 0.9387 +/- 0.0009, test_r2_sensor: 0.8812 +/- 0.0017, test_mae_sensor: 6.4188 +/- 0.0293, test_rmse_sensor: 9.5134 +/- 0.0692, test_mynse_sensor: 0.8806 +/- 0.0017, test_time_sensor: 23.2052 +/- 0.1156, test_throughput_sensor: 44129.0395 +/- 219.9635, 
********************************************************************************
```
Table 11 result:
**1.8928 +/- 0.0149**

<img width="891" height="500" alt="Image" src="https://github.com/user-attachments/assets/18af5b3b-3303-45d6-838d-817944f8e94f" />
I would like to know if there is a problem with the Settings at any step, or rather, how to operate correctly to obtain the values mentioned by your team in the paper? Thanks!

**Comment by atello** [comment]
Dear Zhiyuan,

After looking closely to your question I have observed the following:

1. The results your are trying to replicate correspond to the experiments executed using **Hajgató et al., 2020**, as you can see in the Table 11's caption. However, when you use **gnn_pressure_estimation/scenegenv7.py** to generate the data, you are creating the data with our method, which differs from Hajgató's as our train/val/test set does not have exactly the same distribution. So the results your are obtaining are somewhat comparable to the results presented in Table 10.

2. You should take into account that the configurations of all synthetic datasets are unpublished, unfortunately. The CTown config example is a random solution **guiding you on how to use the generation code**, resulting in an "unpurposed" dataset. This depends on users to define value ranges in their own configuration.

Don't hesitate to ask if you have any further questions.

**Comment by zhiyuan25** [comment]
I'm very glad to receive your reply.  You have corrected my misunderstanding about the functionality of the gnn_pressure_estimation/scenegenv7.py file.   And I also understand that the situation of the dataset is highly related to the configuration, making it difficult to obtain a completely consistent benchmark result. So, in my research, I will focus on the comparison of different models rather than a specific result value. Thank you for your reply!
